lib/time/time.go

```diff
@@ -0,0 +1,515 @@
+// Intentionally exported so that it can be overridden, for example by
+// applications that require their Starlark scripts to be fully deterministic.
+var SleepFunc = time.Sleep
+
+func sleep(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var dur Duration
+	if err := starlark.UnpackPositionalArgs("sleep", args, kwargs, 1, &dur); err != nil {
+		return starlark.None, err
+	}
+
+	SleepFunc(time.Duration(dur))
+	return starlark.None, nil
+}
+
+func parseTime(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		x, location string
+		format      = time.RFC3339
+	)
+	if err := starlark.UnpackArgs("time", args, kwargs, "x", &x, "format?", &format, "location", &location); err != nil {
+		return nil, err
+	}
+
+	if location == "" {
+		t, err := time.Parse(format, x)
+		if err != nil {
+			return nil, err
+		}
+		return Time(t), nil
+	}
+
+	loc, err := time.LoadLocation(location)
+	if err != nil {
+		return nil, err
+	}
+	t, err := time.ParseInLocation(format, x, loc)
+	if err != nil {
+		return nil, err
+	}
+	return Time(t), nil
+}
+
+func fromTimestamp(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var x int
+	if err := starlark.UnpackPositionalArgs("from_timestamp", args, kwargs, 1, &x); err != nil {
+		return nil, err
+	}
+	return Time(time.Unix(int64(x), 0)), nil
+}
+
+func now(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	return Time(NowFunc()), nil
+}
+
+// Duration is a Starlark representation of a duration.
+type Duration time.Duration
+
+func newDuration(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var x Duration
+	if err := starlark.UnpackPositionalArgs("duration", args, kwargs, 1, &x); err != nil {
+		return nil, err
+	}
+	return x, nil
+}
+
+// assert at compile time that Duration implements Unpacker.
+var _ starlark.Unpacker = (*Duration)(nil)
+
+// Unpack is a custom argument unpacker
+func (d *Duration) Unpack(v starlark.Value) error {
+	switch x := v.(type) {
+	case Duration:
+		*d = x
+		return nil
+	case starlark.Int:
+		i, ok := x.Int64()
+		if !ok {
+			return fmt.Errorf("int value out of range (want signed 64-bit value)")
+		}
+		*d = Duration(i)
+		return nil
+	case starlark.String:
+		dur, err := time.ParseDuration(string(x))
+		if err != nil {
+			return err
+		}
+
+		*d = Duration(dur)
+		return nil
+	}
+
+	return fmt.Errorf("cannot convert %s to %s", v.Type(), d.Type())
+}
+
+// String implements the Stringer interface.
+func (d Duration) String() string { return time.Duration(d).String() }
+
+// Type returns a short string describing the value's type.
+func (d Duration) Type() string { return "time.duration" }
```

Comment:
Just in case someone changes the `ModuleName` we should do
```suggestion
func (d Duration) Type() string { return ModuleName+".duration" }
```

Reply:
Agreed. I think we should add this change

Reply:
For consistency, I believe that it should rather remain `time.duration`, otherwise we would need to apply the same change for other modules too. @adonovan WDYT?